tests/protocols/test_http.py

```diff
@@ -492,6 +511,17 @@ async def app(scope: Scope, receive: ASGIReceiveCallable, send: ASGISendCallable
     assert protocol.transport.is_closing()
 
 
+async def test_response_header_splitting(http_protocol_cls: HTTPProtocol):
+    app = Response(b"", headers={"key": "value\r\nCookie: smuggled=value"})
+
+    protocol = get_connected_protocol(app, http_protocol_cls)
+    protocol.data_received(SIMPLE_GET_REQUEST)
+    await protocol.loop.run_one()
+    assert b"HTTP/1.1 500 Internal Server Error" not in protocol.transport.buffer
+    assert b"\r\nCookie: smuggled=value\r\n" not in protocol.transport.buffer
+    assert protocol.transport.is_closing()
```

Comment:
What's this test about? This was already passing before.

Reply:
While working on this PR, I did not find a test that checked that invalid response header values were blocked by uvicorn (preventing, among other issues, header response splitting).

Thus, this test fulfills two purposes for me:

1. It blocks me from implementing "allow horizontal tabs" by removing the regular expression check, either wilfully or inadvertently;
2. It confirms that there is no need to go and change the `h11` implementation for header splitting issues;

In other words, I believed that there was no other test that was covering that bit of functionality, so I decided to add one.